shared/constants/transaction.js

```diff
@@ -192,7 +192,7 @@ export const SMART_TRANSACTION_STATUSES = {
 /**
  * Types that are specific to the transaction approval amount.
  *
- * @typedef {Object} TransactionApprovalAmountType
+ * @typedef {object} TransactionApprovalAmountType
```

Comment:
Like @brad-decker explained [here](https://github.com/MetaMask/metamask-extension/pull/15351#issue-1318770366), an exclusion was added in [this PR](https://github.com/MetaMask/metamask-extension/pull/15293) to prevent lots of casing update for `Object` -> `object`. This removes those exclusions in favor of casing `Object` as `object`.